indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,534 @@
+                emptyTemplate: function(query) { return ""; },
+                display: 'name',
+                callback: {
+                    onInit: function(node) {
+                        this.query = node.val();
+                    },
+                    onClickBefore: function(node, a, item, event) {
+                        validVenue = true;
+                    },
+                    onClickAfter: function(node, a, item, event) {
+                        venueInput.trigger('typeahead:click-on-option', {
+                            item_name: item.name,
+                            venue_id: item.id
+                        });
+                    },
+                    onLayoutBuiltBefore: function(node, query, result, resultHtmlList) {
+                        return updateResultsList(node, resultHtmlList);
+                    },
+                    onSearch: function(node, query) {
+                        this.query = cleanQuery(query);
+                    },
+                    onShowLayout: function(node, query) {
+                        venueInput.next('.highlighter').removeClass('visible');
+                        currentScrollPos = 0;
+                    },
+                    onHideLayout: function(node, query) {
+                        if (justUseOptionSelected) {
+                            venueInput.trigger('typeahead:click-on-custom-option', {item_name: query});
+                        } else if (!validVenue && node.val() && !preventGuessing) {
+                            var result = [];
+                            result = $.grep(venues.data, function(venue) {
+                                return venue.name.toUpperCase() == query.toUpperCase();
+                            });
+                            if (result.length > 0) {
+                                $('#location-venue-' + result[0].id).parent().trigger('click');
+                            } else {
+                                venueInput.trigger('typeahead:click-on-custom-option', {item_name: query});
+                            }
+                        }
+                        preventGuessing = false
+                        validVenue = false;
+                    },
+                    onNavigateBefore: function(node, query, event) {
+                        if (event.keyCode == 9) {
+                            preventGuessing = true
+                        }
+                        resultsWereOpen = node.closest('.typeahead-container').hasClass('result');
+                    },
+                    onResult: function(node, query, result, resultCount, resultCountPerGroup) {
+                        node.parent().find('.just-use-option').toggleClass('active', resultCount === 0);
+                        justUseOptionSelected = resultCount === 0;
+                    }
+
+                }
+            });
+
+            roomInput.typeahead({
```

Comment:
seems pretty similar to the previous typeahead. i didn't compare the code more closely but would it make sense to avoid duplicating some of it if it's really the same?